### PR TITLE
197 reduction operation being executed in parallel with read modify write operation in onesourcegeneratepdf

### PR DESCRIPTION
**Pull request by hdante**
## Change Description

This is version 2 of the original pull request for dealing with issue #197, there are no changes in the final code, so it's a simplification of the patchset. It still should be reviewed though because one comment in the makefile might be misplaced.

- [ ] My PR includes a link to the issue that I am addressing

## Code Quality
- [ ] I have read the Contribution Guide
- [ ] My code follows the code style of this project
- [ ] My code builds (or compiles) cleanly without any errors or warnings
- [ ] My code contains relevant comments and necessary documentation

### Bug Fix Checklist
- [ ] My fix includes a new test that breaks as a result of the bug (if possible)
- [ ] My change includes a breaking change
  - [ ] My change includes backwards compatibility and deprecation warnings (if possible)

### New Feature Checklist
- [ ] I have added or updated the docstrings associated with my feature using the [NumPy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html)
- [ ] I have updated the tutorial to highlight my new feature (if appropriate)
- [ ] I have added unit/End-to-End (E2E) test cases to cover my new feature
- [ ] My change includes a breaking change
  - [ ] My change includes backwards compatibility and deprecation warnings (if possible)

### Documentation Change Checklist
- [ ] Any updated docstrings use the [NumPy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html)

### Build/CI Change Checklist
- [ ] If required or optional dependencies have changed (including version numbers), I have updated the README to reflect this
- [ ] If this is a new CI setup, I have added the associated badge to the README

### Other Change Checklist
- [ ] Any new or updated docstrings use the [NumPy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [ ] I have updated the tutorial to highlight my new feature (if appropriate)
- [ ] I have added unit/End-to-End (E2E) test cases to cover any changes
- [ ] My change includes a breaking change
  - [ ] My change includes backwards compatibility and deprecation warnings (if possible)

### CODE REVIEW

File: CMakeLists.txt
Comment (johannct) [inline CMakeLists.txt:1]: yes the comments are mixed up. How do you want to proceed?

Comment (johannct) [reply CMakeLists.txt:1]: I can edit directly in the branch on github

Comment (hdante) [reply CMakeLists.txt:1]: I'll just amend the commit and send version 3 then.

**Comment by hdante** [comment]
Sent version 3, replacing commit 0de0de097ee61cb81bd79126260eca35d53417a3 with 8bc2ba5ebd2ec8268b777243b520963333389328 (fix location of comments about the thread sanitizer).

File: CMakeLists.txt
Comment (johannct) [reply CMakeLists.txt:1]: I just merged a fix to the file, I think we can go with that no?

**Review by johannct — APPROVED** [review]
LGTM

File: CMakeLists.txt
Comment (hdante) [reply CMakeLists.txt:1]: No, you merged it to the other pull request, which was increased now to 12 commits. I've fixed the second commit and we still have only two in this pull request, that's the idea.

**Comment by johannct** [comment]
ok so it i cleared, you can merged if all is OK.

**Comment by hdante** [comment]
All good now, I don't have merge permission, so you'll need to do it.

**Comment by hdante** [comment]
It worked now, I'll also close issue number #196.

**Pull request merged**